### PR TITLE
Add integration test for DeepL feature

### PR DESCRIPTION
**Pull request by seluianova**
### Short description
In order to provide a stable machine translation feature, we should add tests for DeepL

### Proposed changes
- Make the DeepL API URL configurable
- Add the first portion of tests (to be continued)

### Side effects
I also added and muted one failing test - test_deepl_bulk_machine_translation_api_error.
It seems we don't handle the situation when DeepL works during initialization, but then returns an error when requesting a translation.
image
I will create a separate ticket to fix that.

### Resolved issues
Fixes: #2136

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

**Comment by seluianova** [comment]
Added a check for region.deepl_budget_used after translation.
Added test for translation_exceeds_limit error.

File: tests/deepl_api/deepl_api_test.py [hunk 1]
```diff
@@ -0,0 +1,727 @@
...
+@pytest.mark.django_db
+@pytest.mark.skip(reason="server error handling is currently not implemented")
+def test_deepl_bulk_machine_translation_api_error(
+    load_test_data,
+    login_role_user,
+    settings,
+    httpserver,
+):
+    """
+    This test checks error hanlding when DeepL API returns server error
```
Comment (david-venhoff) [inline tests/deepl_api/deepl_api_test.py:654]:
```suggestion
    This test checks error handling when DeepL API returns server error
```

**Review by david-venhoff — APPROVED** [review]
LGTM!

File: tests/deepl_api/deepl_api_test.py [hunk 2]
```diff
@@ -0,0 +1,727 @@
...
+    :param httpserver: The fixture providing the dummy http server used for faking the DeepL API server
+    :type httpserver: `pytest_httpserver.HTTPServer`
+    """
+    # Test for english messages
+    settings.LANGUAGE_CODE = "en"
+
+    # Setup a mocked DeepL API server with dummy response
+    setup_fake_deepl_api_server(httpserver)
+
+    # Redirect call to the DeepL API to the fake server
```
Comment (PeterNerlich) [inline tests/deepl_api/deepl_api_test.py:126]:
```suggestion
    # Redirect call aimed at the DeepL API to the fake server
```
or
```suggestion
    # Redirect call meant for the DeepL API to the fake server
```

File: tests/deepl_api/deepl_api_test.py [hunk 3]
```diff
@@ -0,0 +1,727 @@
...
```
Comment (PeterNerlich) [inline tests/deepl_api/deepl_api_test.py:232]:
```suggestion
    # Redirect call aimed at the DeepL API to the fake server
```
or
```suggestion
    # Redirect call meant for the DeepL API to the fake server
```

File: tests/deepl_api/deepl_api_test.py [hunk 4]
```diff
@@ -0,0 +1,727 @@
...
```
Comment (PeterNerlich) [inline tests/deepl_api/deepl_api_test.py:343]:
```suggestion
    # Redirect call aimed at the DeepL API to the fake server
```
or
```suggestion
    # Redirect call meant for the DeepL API to the fake server
```

File: tests/deepl_api/deepl_api_test.py [hunk 5]
```diff
@@ -0,0 +1,727 @@
...
+    :param httpserver: The fixture providing the dummy http server used for faking the DeepL API server
+    :type httpserver: `pytest_httpserver.HTTPServer`
+    """
+    # Test for english messages
+    settings.LANGUAGE_CODE = "en"
+
+    # Setup a mocked DeepL API server with dummy response
+    httpserver.expect_request("/v2/translate").respond_with_data("Error", status=500)
+
+    # Redirect call to the DeepL API to the fake server
```
Comment (PeterNerlich) [inline tests/deepl_api/deepl_api_test.py:674]:
```suggestion
    # Redirect call aimed at the DeepL API to the fake server
```
or
```suggestion
    # Redirect call meant for the DeepL API to the fake server
```

**Review by MizukiTemma — APPROVED** [review]
Awesome! Thank you 😸

**Pull request merged**